From 2cf4455685a451dec42f4d98ec0aaf4d6c76a128 Mon Sep 17 00:00:00 2001
From: Richard Danter <radanter@gmail.com>
Date: Mon, 9 May 2022 15:23:45 +0000
Subject: [PATCH 1/2] Revert "firmware: raspberrypi: Fix a leak in
 'rpi_firmware_get()'"

This reverts commit b2f4dd13b211691715e99e28fff9ce9557acc57c.
---
 drivers/firmware/raspberrypi.c | 10 ++--------
 1 file changed, 2 insertions(+), 8 deletions(-)

diff --git a/drivers/firmware/raspberrypi.c b/drivers/firmware/raspberrypi.c
index 1d965c1252ca..8996deadd79b 100644
--- a/drivers/firmware/raspberrypi.c
+++ b/drivers/firmware/raspberrypi.c
@@ -322,18 +322,12 @@ struct rpi_firmware *rpi_firmware_get(struct device_node *firmware_node)
 
 	fw = platform_get_drvdata(pdev);
 	if (!fw)
-		goto err_put_device;
+		return NULL;
 
 	if (!kref_get_unless_zero(&fw->consumers))
-		goto err_put_device;
-
-	put_device(&pdev->dev);
+		return NULL;
 
 	return fw;
-
-err_put_device:
-	put_device(&pdev->dev);
-	return NULL;
 }
 EXPORT_SYMBOL_GPL(rpi_firmware_get);
 
-- 
2.31.1

